tests/ducktape/producer_strategy.py

```diff
@@ -0,0 +1,299 @@
+            
+            # Temporary metrics for timing sections
+            produce_times = []
+            poll_times = []
+            flush_time = 0
+            
+            # Pre-create shared metrics callback to avoid closure creation overhead
+            shared_metrics_callback = None
+            if self.metrics:
+                def shared_metrics_callback(err, msg):
+                    if not err:
+                        # Calculate latency if we have send time
+                        msg_key = msg.key().decode('utf-8', errors='replace') if msg.key() else 'unknown'
+                        latency_ms = 0.0
+                        if msg_key in send_times:
+                            latency_ms = (time.time() - send_times[msg_key]) * 1000
+                            del send_times[msg_key]
+                        self.metrics.record_delivered(latency_ms, topic=msg.topic(), partition=msg.partition())
+            
+            while time.time() - start_time < test_duration:
+                message_value, message_key = message_formatter(messages_sent)
+                try:
+                    # Record sent message for metrics and track send time
+                    if self.metrics:
+                        message_size = len(message_value.encode('utf-8')) + len(message_key.encode('utf-8'))
+                        self.metrics.record_sent(message_size, topic=topic_name, partition=0)
+                        send_times[message_key] = time.time()  # Track send time for latency
+                    
+                    # Produce message
+                    produce_start = time.time()
+                    delivery_future = await producer.produce(
+                        topic=topic_name,
+                        value=message_value,
+                        key=message_key,
+                        on_delivery=shared_metrics_callback
+                    )
+                    produce_times.append(time.time() - produce_start)
+                    pending_futures.append((delivery_future, message_key))  # Store delivery future
+                    messages_sent += 1
+
+                except Exception as e:
+                    if failed_container is not None:
+                        failed_container.append(e)
+                    if self.metrics:
+                        self.metrics.record_failed(topic=topic_name, partition=0)
+                    self.logger.error(f"Failed to produce message {messages_sent}: {e}")
+
+            # Print producer latency metrics
+            producer.print_latency_metrics()
+
+            # Print producer latency stats
+            producer.get_latency_stats()
+
+            # Flush producer
+            flush_start = time.time()
+            await producer.flush(timeout=30)
+            flush_time = time.time() - flush_start
+
+            # Wait for all pending futures to complete (for delivery confirmation only)
+            for delivery_future, message_key in pending_futures:
```

Comment:
Maybe we should track how long we took to put messages into librdkafka in between total e2e delivery times? Ensures we know if it's on the delivery or generation end now and in the future.